public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareTypeSelect.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  isEmailSharingEnabled,
+  PublicDashboardShareType,
+} from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils';
+import { DashboardInteractions } from 'app/features/dashboard-scene/utils/interactions';
+import { AccessControlAction } from 'app/types';
+
+import { useShareDrawerContext } from '../../ShareDrawer/ShareDrawerContext';
+
+import { getAnyOneWithTheLinkShareOption } from './ShareExternally';
+
+const selectors = e2eSelectors.pages.ShareDashboardDrawer.ShareExternally;
+export default function ShareTypeSelect({
+  setShareType,
+  options,
+  value,
+}: {
+  setShareType: (v: SelectableValue<PublicDashboardShareType>) => void;
+  value: SelectableValue<PublicDashboardShareType>;
+  options: Array<SelectableValue<PublicDashboardShareType>>;
+}) {
+  const { dashboard } = useShareDrawerContext();
+  const styles = useStyles2(getStyles);
+
+  const { data: publicDashboard } = publicDashboardApi.endpoints?.getPublicDashboard.useQueryState(
+    dashboard.state.uid!
+  );
+  const [updateAccess, { isLoading }] = useUpdatePublicDashboardAccessMutation();
+
+  const hasWritePermissions = contextSrv.hasPermission(AccessControlAction.DashboardsPublicWrite);
+  const anyOneWithTheLinkOpt = getAnyOneWithTheLinkShareOption();
+
+  const onUpdateShareType = (shareType: PublicDashboardShareType) => {
+    if (!publicDashboard) {
+      return;
+    }
+
+    DashboardInteractions.publicDashboardShareTypeChange({
+      shareType: shareType === PublicDashboardShareType.EMAIL ? 'email' : 'public',
+    });
+
+    const req = {
+      dashboard,
+      payload: {
+        ...publicDashboard!,
+        share: shareType,
+      },
+    };
+
+    updateAccess(req);
+  };
+
+  return (
+    <div>
+      <Stack justifyContent="space-between">
+        <Label description={value.description}>
+          <Trans i18nKey="public-dashboard.share-configuration.share-type-label">Link access</Trans>
+        </Label>
+        {isLoading && <Spinner />}
+      </Stack>
+      <Stack direction="row" gap={1} alignItems="center">
+        {isEmailSharingEnabled() ? (
+          <Select
+            data-testid={selectors.shareTypeSelect}
+            options={options}
+            value={value}
+            disabled={!hasWritePermissions}
+            onChange={(v) => {
+              setShareType(v);
+              onUpdateShareType(v.value!);
+            }}
+            className={styles.select}
+          />
+        ) : (
+          <>
+            {toIconName(anyOneWithTheLinkOpt.icon) && <Icon name={toIconName(anyOneWithTheLinkOpt.icon)!} />}
+            <Text>{anyOneWithTheLinkOpt.label}</Text>
+          </>
+        )}
+        <Text element="p" variant="bodySmall" color="disabled">
+          <Trans i18nKey="public-dashboard.share-configuration.access-label">can access</Trans>
+        </Text>
+      </Stack>
+    </div>
+  );
+}
+
+const getStyles = () => {
+  return {
+    select: css({
```

Comment:
When the first option is selected, the difference in colors is barely noticeable when hovering over options.

Reply:
This problem is probably related to using a Select component for only two options instead of a radio button. We don't want to block this PR because there are dependencies but we are going to work on this soon.